pkg/sql/opt/optbuilder/insert.go

```diff
@@ -641,11 +641,13 @@ func (mb *mutationBuilder) buildInsert(returning tree.ReturningExprs) {
 	// check constraint, refer to the correct columns.
 	mb.disambiguateColumns()
 
+	preCheckScope := mb.outScope
```

Comment:
Add comment here explaining why this is needed.

Reply:
Done.